magefile.go

```diff
@@ -0,0 +1,256 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+var (
+	tools = []string{
+		"github.com/bufbuild/buf/cmd/buf",
+		"github.com/bufbuild/buf/cmd/protoc-gen-buf-breaking",
+		"github.com/bufbuild/buf/cmd/protoc-gen-buf-lint",
+		"github.com/golangci/golangci-lint/cmd/golangci-lint",
+		"github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway",
+		"github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2",
+		"golang.org/x/tools/cmd/cover",
+		"golang.org/x/tools/cmd/goimports",
+		"google.golang.org/grpc/cmd/protoc-gen-go-grpc",
+		"google.golang.org/protobuf/cmd/protoc-gen-go",
+	}
+
+	Default = Build
+)
+
+// Bootstrap installs tools required for development
+func Bootstrap() error {
+	fmt.Println("Bootstrapping tools...")
+	if err := os.Chdir("_tools"); err != nil {
+		return fmt.Errorf("failed to change dir: %w", err)
+	}
+
+	// create module if go.mod doesnt exist
+	if _, err := os.Stat("go.mod"); os.IsNotExist(err) {
+		cmd := exec.Command("go", "mod", "init", "tools")
+		if err := cmd.Run(); err != nil {
+			return err
+		}
+	}
+
+	goInstall := sh.RunCmd("go", "install", "-v")
+	return goInstall(tools...)
+}
+
+// Build builds the project
+func Build() error {
+	mg.Deps(Prep)
+	fmt.Println("Building...")
+
+	buildDate, err := sh.Output("date", "-u", "+%Y-%m-%dT%H:%M:%SZ")
+	if err != nil {
+		return fmt.Errorf("failed to get build date: %w", err)
+	}
```

Comment:
```suggestion
	buildDate := time.Now().UTC().Format(time.RFC3339)
```

It should be a tad bit more portable too.

Reply:
i forget, now that we are in Go the possibilities are endless 🌅

Reply:
It is a very cool feature of the way these kinds of tools are taking things.
The same applies for things like Pulumi and Dagger.